### PR TITLE
Add pip install xonsh [full] and rewrite install doc

### LINKED ISSUE #3622: Please, add 'prompt_toolkit' to general requirements.
I'm having errors in 'abbrevs' and a lot of things printed at the terminal startup.
I use Manjaro Linux and neither the system package, nor the "Arch User Repository" package, nor the package in the "pip" requires the installation of "prompt_toolkit" by DEFAULT.
If possible, as it is undoubtedly an important package, add as required dependency.
In a last look at all the documentation, I found this syntax in the installation instructions for linux: "xonsh[ptk,linux]". I would like to say that, even for those who have a basic knowledge of python, this syntax is a bit obscure. If possible, it would be good to clarify the point.
Thank you very much for this wonderful project.

### PR DESCRIPTION
**Pull request by bobhy**
Resolves #3622 

Adds extras tag [full], which installs xonsh with pygments, prompt-toolkit and distro, with the intent that that's what most users will want.

Also update installation section of docs to include this option, but take the opportunity for a thorough rewrite:  
collapse separate linux, osx and windows guides into one cross-platform install and config flow; add doc for the 2 startup wizards, modernize the Windows section to prefer Windows Terminal as your getting-started config.    

These doc changes are opinionated and potentially controversial, which is not my intent.  So I'm prepared to redo based on feedback.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -440,5 +440,5 @@ def make_events():
 def setup(app):
     from xonsh.pyghooks import XonshConsoleLexer
 
-    app.add_lexer("xonshcon", XonshConsoleLexer)
+    app.add_lexer("xonshcon", XonshConsoleLexer())
```
Comment (scopatz) [inline docs/conf.py:443]: Doesn't this need to be a class, not an instance?

Comment (scopatz) [reply docs/conf.py:443]: Or is it some weird magic thing that Pygments does?

File: docs/jupyter.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Xonsh kernel for Jupyter
+========================
+
+[placeholder]
```
Comment (scopatz) [inline docs/jupyter.rst:4]: This should be filled out or removed

File: docs/packages.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
+Via Package Manager
+===================
+
+You can install xonsh using ``conda``, ``pip`` or the package manager for 
+your operating system distribution.
+
+For the fullest interactive user experience, these additional packages should also be installed:
+
+  :prompt-toolkit: for command completion, configurable key bindings and especially multi-line line editing.
+  :pygments: for xonsh and python syntax-specific highlighting
```
Comment (scopatz) [inline docs/packages.rst:10]: `Python`

File: docs/packages.rst [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+If you are operating in a specialized or restricted environment, you can install just the xonsh package, as 
+described in `fewer prerequisites`_
+
+
+**conda:**
+
+.. code-block:: console
+
+    $ conda config --add channels conda-forge
+    $ conda install pygments prompt-toolkit setproctitle xonsh
```
Comment (scopatz) [inline docs/packages.rst:23]: I think the conda-forge package includes those as deps, so this just needs to be `conda install xonsh`

File: docs/packages.rst [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+.. code-block:: console
+
+    $ pip install pygments prompt-toolkit setproctitle xonsh
+
+As a shortcut (for pip only), you can specify all of the above with `extras` syntax:
+
+.. code-block:: console
+
+    $ pip install xonsh[full]
```
Comment (scopatz) [inline docs/packages.rst:37]: probably should start with this, for clarity

File: docs/packages.rst [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+The above ``pip`` commands may have to be spelled ``pip3`` or ``sudo pip3`` if you are not installing in a virtual environment.
+
+**source:** Pip can also install the most recent xonsh source code from the 
+`xonsh project repository <https://github.com/xonsh/xonsh>`_.
+
+.. code-block:: console
+
+    $ pip install pygments prompt-toolkit setproctitle https://github.com/xonsh/xonsh/archive/master.zip
```
Comment (scopatz) [inline docs/packages.rst:46]: I think we should recommend people clone the repo and then `pip install --user .`  rather than use the master archive

File: docs/packages.rst [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+Fewer Prerequisites
+--------------------
+
+A design goal of Xonsh is to run in any environment that supports a (supported) Python interpreter, you 
+can install just the ``xonsh`` package (using any package manager).
+
+.. code-block:: console
+  
+  pip install xonsh
```
Comment (scopatz) [inline docs/packages.rst:79]: four space indent

File: docs/platform-issues.rst [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+
+    unset module
+    unset scl
+
+
+Default Ubuntu .bashrc breaks Foreign Shell Functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Xonsh supports importing functions from foreign shells using the
+`ForeignShellFunctionAlias` class, which calls functions as if they were
```
Comment (scopatz) [inline docs/platform-issues.rst:38]: double backticks needed in rst :wink:

File: docs/platform-issues.rst [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+aliases. This is implemented by executing a command that sources the file
+containing the function definition and then immediately calls the function with
+any necessary arguments.
+
+The default user `~/.bashrc` file in Ubuntu 15.10 has the following snippet at
```
Comment (scopatz) [inline docs/platform-issues.rst:43]: double backticks

File: docs/platform-issues.rst [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+yields the same problem.
+
+
+MacOS, OSX
+----------
+
+readline
+^^^^^^^^
+
+[ed note: This recommendation seems to be `out of date <https://pypi.org/project/gnureadline/>`_.
```
Comment (scopatz) [inline docs/platform-issues.rst:67]: Can you rephrase to make it seem less like the docs are contradicting themselves. Maybe soften the lang below?

File: docs/platform-issues.rst [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+
+Tab completion
+^^^^^^^^^^^^^^
+Xonsh has support for using bash completion files on the shell, to use it you need to install 
+the bash-completion package. 
+The regular bash-completion package uses v1 which mostly works, but `occasionally has rough edges <https://github.com/xonsh/xonsh/issues/2111>`_ so we recommend using bash-completion v2.
+
+Bash completion comes from <https://github.com/scop/bash-completion> which suggests you use a package manager to install it, this manager will also install a new version of bash without affecting  /bin/bash. Xonsh also needs to be told where the bash shell file that builds the completions is, this has to be added to $BASH_COMPLETIONS. The package includes completions for many Unix commands.
+
+Common packaging systems for MacOs include
```
Comment (scopatz) [inline docs/platform-issues.rst:104]: consistent spelling of macOS

File: docs/xonshrc.rst [hunk 12]
```diff
@@ -1,13 +1,160 @@
...
+* Xonsh function defintions
+* `Alias definitions <aliases.html>`_, many of which invoke the above functions with specified arguments.
 
-Xonsh also allows you to have a run control file in your home directory. It can either be directly in the home directory at ``~/.xonshrc`` or for XDG compliance at ``~/.config/rc.xsh``. The options set in the local ``xonshrc`` only apply to the current user and will override any conflicting settings set in the system-wide control file.
+The system-wide ``xonshrc`` file controls options that are applied to all users of Xonsh on a given system.  
+You can create this file in ``/etc/xonshrc`` for Linux and OSX and in ``%ALLUSERSPROFILE%\xonsh\xonshrc`` on Windows.
 
-These files are written in the xonsh language, of course. They are executed exactly once
-at startup. The following is a real-world example of such a file.
+Xonsh also allows a per-user  run control file in your home directory, either
```
Comment (scopatz) [inline docs/xonshrc.rst:15]: double space between `per-user` and `run` should be single space

File: docs/xonshrc.rst [hunk 13]
```diff
@@ -1,13 +1,160 @@
...
+directly in the home directory at ``~/.xonshrc`` or, for XDG compliance, at ``~/.config/rc.xsh``. 
+The options set per user override settings in the system-wide control file.
+
+Xonsh provides 2 wizards to create your own "xonshrc".  ``xonfig web`` provides basic settings, and ``xonfig wizard``
+steps you through all the available options.
+
+Xonfig web
```
Comment (scopatz) [inline docs/xonshrc.rst:22]: lowercase `x` to be consistent with heading below

File: docs/xonshrc.rst [hunk 14]
```diff
@@ -1,13 +1,160 @@
...
+:Colors: shows the  color themes built into Xonsh.  
+  Simply click on a sample to select it.  Although color names are standardized across various terminal applications, 
+  their actual appearance is not and do vary widely.  Seeing is believing! 
+:Prompts: shows various sample prompts.  It is recommended to select one but to then edit 
+  the ``xonshrc`` file to further refine your prompt.
+:Xontribs: are community-contributed add-ins often used to enhance command completion and line editing, 
+  but can affect any aspect of Xonsh behavior.  
+  Choose one or more to suit your needs but note that they will require installation of additional
+  packages.  You can extend Xonsh by `writing your own xontrib <tutorial_xontrib.html>`_, and are invited/urged to do so!
+:Save: Click to write the configuration choices to your `~/.xonshrc`. This will add a few tagged lines to your run control file, but will not
```
Comment (scopatz) [inline docs/xonshrc.rst:47]: double backticks

File: docs/xonshrc.rst [hunk 15]
```diff
@@ -1,13 +1,160 @@
...
+  Simply click on a sample to select it.  Although color names are standardized across various terminal applications, 
+  their actual appearance is not and do vary widely.  Seeing is believing! 
+:Prompts: shows various sample prompts.  It is recommended to select one but to then edit 
+  the ``xonshrc`` file to further refine your prompt.
+:Xontribs: are community-contributed add-ins often used to enhance command completion and line editing, 
+  but can affect any aspect of Xonsh behavior.  
+  Choose one or more to suit your needs but note that they will require installation of additional
+  packages.  You can extend Xonsh by `writing your own xontrib <tutorial_xontrib.html>`_, and are invited/urged to do so!
+:Save: Click to write the configuration choices to your `~/.xonshrc`. This will add a few tagged lines to your run control file, but will not 
+  overwrite it completely, so you can run `xonfig web` at any time.
```
Comment (scopatz) [inline docs/xonshrc.rst:48]: double backticks

File: docs/xonshrc.rst [hunk 16]
```diff
@@ -1,13 +1,160 @@
...
+xonfig wizard
+--------------
+
+This imports settings and tools you have defined in your existing (ordinary) shell such as ``bash``.  
+It also walks you through setting all known environment variables and xontribs 
+in a question-and-answer format:
+
+.. code-block:: xonshcon
+
+  $ xonfig wizard
```
Comment (scopatz) [inline docs/xonshrc.rst:59]: four space indent in this block

**Comment by scopatz** [comment]
Thanks for the awesome effort here @bobhy!

File: docs/conf.py (hunk 1 above)
Comment (bobhy) [reply docs/conf.py:443]: It's what I needed to do to get Sphinx to stop complaining about code-block xonshcon.

Comment (scopatz) [reply docs/conf.py:443]: Ahh cool that's great!

File: docs/packages.rst (hunk 4 above)
Comment (bobhy) [reply docs/packages.rst:23]: OK, I'll doc it.
But how do you conda install without the extra packages, e.g for emacs?
If you can't (which is what it looks like), why can't we just do it the same way for vanilla pip?

File: docs/packages.rst (hunk 6 above)
Comment (bobhy) [reply docs/packages.rst:46]: But that would not be a source install?  I admit I am fuzzy on the difference, or the tradeoffs...

File: docs/jupyter.rst (hunk 2 above)
Comment (bobhy) [reply docs/jupyter.rst]: Removed (for now).  I don't have a reliable install procedure to document!  I got this to work once (and ran into a runtime issue), but now I cannot get setup to recognize that it should install the jupyter kernel, either in pip or conda.  A problem for another day...

**Comment by bobhy** [comment]
Better?
I can't get setup to install jupyter kernel, have temporarily left it out of the docs.

**Comment by scopatz** [comment]
Thanks!
